src/handlers/compositor.rs

```diff
@@ -297,13 +297,42 @@ impl CompositorHandler for State {
             &self.niri.cursor_manager.cursor_image(),
             CursorImageStatus::Surface(s) if s == &root_surface
         ) {
+            with_states(surface, |states| {
```

Comment:
Guess this should only happen if `surface == root_surface`

Reply:
right, while unlikely the surface could be part of a surface tree